src/sentry/discover/compare_tables.py

```diff
@@ -0,0 +1,261 @@
+import logging
+from enum import Enum
+from typing import TypedDict, cast
+
+import sentry_sdk
+
+from sentry.discover.arithmetic import is_equation
+from sentry.discover.dataset_split import _get_equation_list, _get_field_list
+from sentry.discover.translation.mep_to_eap import (
+    QueryParts,
+    translate_columns,
+    translate_mep_to_eap,
+)
+from sentry.models.dashboard import Dashboard
+from sentry.models.dashboard_widget import DashboardWidget, DashboardWidgetQuery
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+from sentry.search.eap.types import EAPResponse, SearchResolverConfig
+from sentry.search.events.types import SAMPLING_MODES, EventsResponse, SnubaParams
+from sentry.snuba import metrics_enhanced_performance, spans_rpc
+
+logger = logging.getLogger(__name__)
+
+
+class CompareTableResult(Enum):
+    BOTH_FAILED = "both_requests_failed"
+    FIELD_NOT_FOUND = "field_not_found"
+    METRICS_FAILED = "metrics_failed"
+    NO_FIELDS = "no_fields"
+    NO_PROJECT = "no_project"
+    PASSED = "passed"
+    EAP_FAILED = "eap_failed"
+
+
+class CompareTableResultDict(TypedDict):
+    passed: bool
+    reason: CompareTableResult
+    fields: list[str] | None
+    widget_query: DashboardWidgetQuery
+    mismatches: list[str] | None
+
+
+mismatched_tags_count: dict[str, int] = {}
+
+
+def compare_table_results(metrics_query_result: EventsResponse, eap_result: EAPResponse):
+    eap_data_row = eap_result["data"][0] if len(eap_result["data"]) > 0 else {}
+    metrics_data_row = (
+        metrics_query_result["data"][0] if len(metrics_query_result["data"]) > 0 else {}
+    )
+    metrics_fields = metrics_query_result["meta"]["fields"]
+
+    mismatches = []
+    no_metrics_data = len(metrics_data_row) == 0
+
+    # if there's no metrics data we know there are mismatches,
+    # we will check the EAP data for the names of the mismatched fields
+    if no_metrics_data:
+        for field, data in metrics_fields.items():
+            if is_equation(field):
+                continue
+            mismatches.append(field)
+            mismatched_tags_count[field] = mismatched_tags_count.get(field, 0) + 1
+        return (len(mismatches) == 0, mismatches)
```

Comment:
I'm not sure I follow 👀 
If there is no metrics data, it doesn't tell us any information about mismatches. it just means there is no metrics/transaction data (which means, probably no span data for this query either), so we can just skip tracking mismatches here. I'd introduce a "NO_DATA" enum to your `CompareTableResult` and the reason as that for this scenario.
```suggestion
    if no_metrics_data:
        return (False, [], CompareTableResult.NO_DATA)  # update this tuple to return the reason, for example
```